be/src/runtime/descriptors.cpp

```diff
@@ -148,6 +148,10 @@ HdfsPartitionDescriptor::HdfsPartitionDescriptor(const TDeltaLakeTable& thrift_t
           _location(thrift_partition.location.suffix),
           _thrift_partition_key_exprs(thrift_partition.partition_key_exprs) {}
 
+HdfsPartitionDescriptor::HdfsPartitionDescriptor(const TIcebergTable& thrift_table,
+                                                 const THdfsPartition& thrift_partition)
+        : _thrift_partition_key_exprs(thrift_partition.partition_key_exprs) {}
```

Comment:
Error handling: In the constructor for HdfsPartitionDescriptor::HdfsPartitionDescriptor(const TIcebergTable& thrift_table, const THdfsPartition& thrift_partition), only thrift_partition.partition_key_exprs is used even though both parameters are passed to the constructor. There should either be some error handling if thrift_table is not supposed to be null, or you might want to remove it from the parameters list if it isn't needed.

Reply:
It's historical reason, TIcebergTable is not used, but it's helpful to overload function. just keep style with hive/hudi/deltalake, also we can deal it better.